components/private_channel/BUILD.gn

```diff
@@ -0,0 +1,32 @@
+import("//brave/components/private_channel/buildflags.gni")
+import("//build/buildflag_header.gni")
+
+buildflag_header("buildflags") {
```

Comment:
it's better to put this in components/private_channel/buildflags/BUILD.gn because this file should have `assert(private_channel_enabled)` so it doesn't get set as a dep without the buildflag enabled like https://github.com/brave/brave-core/pull/5765/files#r508030279

Reply:
- [x] add `assert(private_channel_enabled)` and refactor location of `BUILD`